### PR TITLE
General code cleanup

### PR DESCRIPTION
**Pull request by chromium-52**
A few misc. changes and code cleanup:
1) Abstract `isAdminCheck()` and rename to `assertAdminOrSuperAdmin()` - create abstract class `AbstractProcessor`
2) Clean up code (remove unused imports, fix typos)
3) Resolve 2 TODO items
4) Update email templates

Email screenshots:
Account Deactivated:
account_deactivated

Email Change Confirmation:
email_change_confirm

Team Invite Email:
team_invite

Neighborhood Notification Email:
neighborhood_notification

Password Change Confirmation Email:
password_change_confirm

Password Change Request Email:
(Template changed in https://github.com/Code-4-Community/speak-for-the-trees-backend-v2/pull/109)

Welcome Email:
welcome_email

### CODE REVIEW

**Review by Tvcz — APPROVED** [review]
Abstract admin check - nice. 👍

File: api/src/main/java/com/codeforcommunity/rest/subrouter/SiteRouter.java [hunk 1]
```diff
@@ -44,12 +44,12 @@ private void handleGetSiteRoute(RoutingContext ctx) {
...
 
-  private void registerGetStewardshipActivites(Router router) {
+  private void registerGetStewardshipActivities(Router router) {
     Route getStewardshipActivities = router.get("/:site_id/stewardship_activities");
-    getStewardshipActivities.handler(this::handleGetStewardshipActivites);
+    getStewardshipActivities.handler(this::handleGetStewardshipActivities);
   }
 
-  private void handleGetStewardshipActivites(RoutingContext ctx) {
+  private void handleGetStewardshipActivities(RoutingContext ctx) {
```
Comment (SofieDunt) [inline api/src/main/java/com/codeforcommunity/rest/subrouter/SiteRouter.java:52]: 👍

File: service/src/main/java/com/codeforcommunity/processor/AbstractProcessor.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+   * Throws an exception if the user is not an admin or super admin.
+   *
+   * @param level the privilege level of the user calling the route
+   */
+  protected void assertAdminOrSuperAdmin(PrivilegeLevel level) {
+    if (!(level.equals(PrivilegeLevel.ADMIN) || level.equals(PrivilegeLevel.SUPER_ADMIN))) {
+      throw new AuthException("User does not have the required privilege level.");
+    }
+  }
+}
```
Comment (SofieDunt) [inline service/src/main/java/com/codeforcommunity/processor/AbstractProcessor.java:17]: 😍

**Review by SofieDunt — APPROVED** [review]
💯

**Pull request merged**